DOCSP-31807: includeResultMetadata option behavior changes for findOneAnd... methods (#750)
mongodb/docs-node: source/fundamentals/crud/compound-operations.txt

   Starting in version 5.7, you can set the ``includeResultMetadata``
   setting in the ``options`` object to specify the return type for each
   of these methods.
   

>    Starting in version 6.0, the setting defaults to

Reviewer: Q: This might be a larger team discussion, but I'm always unsure about whether to include "Starting in..." language like this, since the change will only be published in that version anyway. Do you think it's helpful to users?

Author: Not making any changes, as we should have a team discussion- will address in a cleanup ticket later depending on discussion outcome

>    Starting in version 6.0, the setting defaults to

Reviewer: S: Just for clarity

```suggestion
   Starting in version 6.0, this setting defaults to
```

>    ``false``, which means each method returns either the matched document
>    or ``null`` if no document is matched. If you set

Reviewer: S: To reduce ambiguity. Definitely open to alternatives!

```suggestion
   ``false``, which means each method returns the matched document,
   if one is found, or ``null``. If you set
```

Author: edited slightly

Reviewer: not a blocking change, but there's still ambiguity around what the method returns and under which conditions. i.e., this sentence could also mean "If no document is matched, the method returns either the matched document or null." 

putting the condition in a parenthetical tied to the first case (what i suggested above) is one way to handle this, but there are probably others.

Reviewer: Oddly, "parenthetical" can mean any [interjection of qualifying or explanatory information](https://www.collinsdictionary.com/us/dictionary/english/parenthetical)--even when it's offset by commas. That's what I meant here.

      { _id: 1, x: "on" }

   The following code shows the return type for ``findOneAndDelete()``
   operations depending on the value of the ``includeResultMetadata`` option:

   .. code-block:: js

>       // default, returns { _id: 1, x: 'on' }

Reviewer: S: for clarity, something like:

```suggestion
      // default behavior: returns { _id: 1, x: 'on' }
```
